### DISCUSSION TITLE
Is Storybook restricted to only JS/TS ecosystems?

### DISCUSSION THREAD

**Discussion by NickLarsenNZ** [storybookjs/storybook#23074] [Help]
Is it possible to use storybook some way with other languages and frameworks such as [Leptos](https://leptos.dev/) (Rust)?

**Comment by f3bw** [comment]
Storybook is primarily designed for building UI components in JavaScript frameworks like React, Vue, and Angular. It provides a development environment where you can develop, test, and showcase your components in isolation. While Storybook itself is focused on JavaScript-based frameworks, it might still be possible to use it with other languages and frameworks like Leptos (Rust) to some extent, depending on the specific requirements and capabilities of the framework.

Here are a few considerations if you want to use Storybook with Leptos or any other non-JavaScript framework:

1. Integration: Storybook relies on a JavaScript runtime and typically requires a JavaScript build system. If Leptos (Rust) can generate a JavaScript-compatible output or has bindings to enable JavaScript interoperability, you might be able to integrate Storybook into your development workflow.

2. Component Rendering: Storybook's core functionality revolves around rendering and showcasing UI components. You would need to find a way to render your Leptos components in a JavaScript environment. 

3. Configuration: Storybook has a configuration file where you define various settings and options for your components. If you're using Leptos (Rust) with Storybook, you would need to configure Storybook to work with your specific framework and define the necessary build steps, compilers, or bundlers required for your Rust code.

4. Tooling and Ecosystem: Storybook has a rich ecosystem and supports various addons, plugins, and tools specifically tailored for JavaScript frameworks. You would need to assess whether these tools can be adapted or replaced with equivalents in the Leptos (Rust) ecosystem.

Overall, while Storybook is primarily focused on JavaScript frameworks, it's possible to explore integrating it with Leptos or other non-JavaScript frameworks. However, the level of effort required would depend on the compatibility and interop capabilities between JavaScript and your chosen framework. You may need to invest additional effort in adapting Storybook to work smoothly with Leptos (Rust) and leverage the existing ecosystem of tools and add-ons.
